### PR TITLE
Fix `DataFrame._from_arrays` and introduce validations

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Fixes: #17111  This PR fixes `DataFrame._from_arrays` to properly access `ndim` attribute and also corrects two validations in `Series` & `DataFrame` constructors.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -781,9 +781,15 @@ def __init__(
                 )
         elif isinstance(data, ColumnAccessor):
             raise TypeError(
-                "Use cudf.Series._from_data for constructing a Series from "
+                "Use cudf.DataFrame._from_data for constructing a Series from "
```
Comment (Matt711) [inline python/cudf/cudf/core/dataframe.py:784]: Should that be `DataFrame`?
```suggestion
                "Use cudf.DataFrame._from_data for constructing a DataFrame from "
```

File: python/cudf/cudf/core/dataframe.py [hunk 2]
```diff
@@ -781,9 +781,15 @@ def __init__(
...
             raise TypeError(
-                "Use cudf.Series._from_data for constructing a Series from "
+                "Use cudf.DataFrame._from_data for constructing a DataFrame from "
                 "ColumnAccessor"
             )
+        elif isinstance(data, ColumnBase):
+            raise TypeError(
+                "Use cudf.DataFrame._from_arrays for constructing a Series from "
+                "ColumnAccessor or Use cudf.DataFrame._from_data by passing a dict "
+                "of column name and column as key-value pair."
```
Comment (galipremsagar) [inline python/cudf/cudf/core/dataframe.py:791]:
```suggestion
                "Use cudf.DataFrame._from_arrays for constructing a DataFrame from "
                "ColumnAccessor or Use cudf.DataFrame._from_data by passing a dict "
                "of column name and column as key-value pair."
```

File: python/cudf/cudf/core/dataframe.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py]: Yup, thanks for the catch!

File: python/cudf/cudf/core/dataframe.py [hunk 3]
```diff
@@ -781,9 +781,15 @@ def __init__(
...
             raise TypeError(
-                "Use cudf.Series._from_data for constructing a Series from "
+                "Use cudf.DataFrame._from_data for constructing a DataFrame from "
                 "ColumnAccessor"
             )
+        elif isinstance(data, ColumnBase):
+            raise TypeError(
+                "Use cudf.DataFrame._from_arrays for constructing a DataFrame from "
+                "ColumnAccessor or Use cudf.DataFrame._from_data by passing a dict "
+                "of column name and column as key-value pair."
```
Comment (Matt711) [inline python/cudf/cudf/core/dataframe.py:791]:
```suggestion
                "Use cudf.DataFrame._from_arrays for constructing a DataFrame from "
                "ColumnBase or Use cudf.DataFrame._from_data by passing a dict "
                "of column name and column as key-value pair."
```

**Review by Matt711 — APPROVED** [review]
So would this example `cudf.DataFrame(cudf.Series([3])._column)` now raise a TypeError?

**Comment by galipremsagar** [comment]
> So would this example `cudf.DataFrame(cudf.Series([3])._column)` now raise a TypeError?

Yup.

**Pull request merged**